backend/common.go

```diff
@@ -39,7 +39,10 @@ func (s *AppInstanceSettings) HTTPClientOptions() (httpclient.Options, error) {
 		return httpclient.Options{}, err
 	}
 
-	return httpSettings.HTTPClientOptions(), nil
+	opts := httpSettings.HTTPClientOptions()
+	setCustomOptionsFromHTTPSettings(&opts, httpSettings)
```

Comment:
Can `setCustomOptionsFromHTTPSettings` be called from within `httpSettings.HTTPClientOptions()` instead?

Reply:
I did that first, but since `JSONDataFromHTTPClientOptions` and `SecureJSONDataFromHTTPClientOptions` operates on `httpclient.Options` it made more sense to me not giving `HTTPSettings` the responsibility for setting custom options.